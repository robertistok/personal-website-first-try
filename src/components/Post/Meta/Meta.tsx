import React from "react";
import moment from "moment";
import styles from "./Meta.module.scss";

interface MetaProps {
  date: string;
}

const Meta: React.FunctionComponent<MetaProps> = ({ date }: MetaProps): React.ReactElement => (
  <div className={styles["meta"]}>
    <p className={styles["meta__date"]}>Published {moment(date).format("D MMM YYYY")}</p>
  </div>
);

export default Meta;
