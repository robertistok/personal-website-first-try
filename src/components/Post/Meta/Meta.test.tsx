import React from "react";
import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";

import Meta from "./Meta";

describe("<Meta />", () => {
  const props = {
    date: "2016-09-01"
  };

  it("renders", () => {
    shallow(<Meta {...props} />);
  });

  it("renders and matches snapshot", () => {
    const wrapper = shallow(<Meta {...props} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
