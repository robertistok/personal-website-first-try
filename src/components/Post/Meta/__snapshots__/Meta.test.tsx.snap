// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Meta /> renders and matches snapshot 1`] = `
<div
  className="meta"
>
  <p
    className="meta__date"
  >
    Published 
    1 Sep 2016
  </p>
</div>
`;
